### PR TITLE
feat(jdbc): native array reading support

### LINKED ISSUE #5278: jdbc time data type does not support sub second
### What happens?

When selecting a time with a sub second component the JDBC driver throws an exception when creating the result set, which then returns null from the catch statement block.

The JDBC driver DuckDbResultSet class uses the java.sql.Time class. In 

```java
    public Time getTime(int var1) throws SQLException {
        String var2 = this.getLazyString(var1);
        if (var2 == null) {
            return null;
        } else {
            try {
                return Time.valueOf(this.getLazyString(var1));
            } catch (Exception var4) {
                return null;
            }
        }
    }
```

The `Time.valueOf(...)` call passes a String to be parsed into a Time value. As the java.sql.Time javadoc states:
```
public static [Time] valueOf([String  s)
Converts a string in JDBC time escape format to a Time value.
Parameters:
s - time in format "hh:mm:ss"
Returns:
a corresponding Time object
```

Passing a time value with a milli/nano second component with the form 'hh:mm:ss.SSS' results in an exception which causes a null to be returned.

### To Reproduce
From DBeaver execute the SQL
```sql
select current_time
```
Returns null.

Executing 
```sql
select cast(current_time as VARCHAR(30))
```
Returns '22:39:01.348'

Executing directly from a java program, (although more complicated to reproduce here) results in the same problem.

OS:
Windows

DuckDB Version:
0.51.0

DuckDB Client:
DBeaver and Java JDBC

Full Name:
Dorian Jones

Affiliation:
None

Have you tried this on the latest master branch?
Yes
Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?
Yes

### LINKED ISSUE #7218: NullPointerException when reading ResultSet.getInt() from GENERATE_SERIES query
### What happens?

A query that reads `ResultSet.getInt()` from `GENERATE_SERIES()` with bind variable arguments fails with a `NullPointerException`

### To Reproduce

```java
try (Statement s = connection.createStatement()) {
    s.executeUpdate("drop table if exists t");
    s.executeUpdate("create table t (i int)");

    try (PreparedStatement p = connection.prepareStatement("select generate_series.generate_series from generate_series(?, ?) order by 1")) {
        p.setInt(1, 0);
        p.setInt(2, 1);

        try (ResultSet rs = p.executeQuery()) {
            while (rs.next())
                println(rs.getInt(1));
        }
    }
}
```

This produces:

```
Exception in thread "main" java.lang.NullPointerException: Cannot load from object array because "this.current_chunk[...].varlen_data" is null
	at org.duckdb.DuckDBResultSet.getLazyString(DuckDBResultSet.java:248)
	at org.duckdb.DuckDBResultSet.getObject(DuckDBResultSet.java:213)
	at org.duckdb.DuckDBResultSet.getInt(DuckDBResultSet.java:325)
	at org.jooq.testscripts.JDBC.main(JDBC.java:55)
```

### OS:

Microsoft Windows [Version 10.0.22621.1555]

### DuckDB Version:

v0.7.2-dev2430

### DuckDB Client:

JDBC

### Full Name:

Lukas Eder

### Affiliation:

Data Geekery / jOOQ

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

### PR DESCRIPTION
**Pull request by Mause**
Most of the work in this PR was moving various methods from DuckDBResultSet into DuckDBVector.

From here it should be straightforward to add Struct and Map support

Fixes part of https://github.com/duckdb/duckdb/discussions/7173, and incidentally, fixes https://github.com/duckdb/duckdb/issues/7218 and fixes https://github.com/duckdb/duckdb/issues/5278

### CODE REVIEW

File: tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java [hunk 1]
```diff
@@ -3211,13 +3219,36 @@ public static void test_union() throws Exception {
...
+				assertTrue(rs.next());
+				assertEquals(arrayToList(rs.getArray(1)), singletonList(1));
+			}
+			try (ResultSet rs = statement.executeQuery("select unnest([[1], [42, 69]])")) {
+				assertTrue(rs.next());
+				assertEquals(arrayToList(rs.getArray(1)), singletonList(1));
+				assertTrue(rs.next());
+				assertEquals(arrayToList(rs.getArray(1)), Arrays.asList(42, 69));
+			}
+			try (ResultSet rs = statement.executeQuery("select unnest([[[42], [69]]])")) {
```
Comment (Mytherin) [inline tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java:3233]: I would like to see some more tests with nested lists, lists of strings, empty lists, etc. Perhaps we could add the test with `test_all_types` while excluding the columns/types not yet supported? That should cover a lot of the potential edge cases here.

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Looks good to me - one request for more tests:

File: tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java (hunk 1 above)
Comment (Mause) [reply tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java:3252]: Okay, added the test_all_types() test - not sure if this one is still worth keeping?

**Comment by Mytherin** [comment]
Thanks! LGTM

**Pull request merged**